### PR TITLE
feat(python): pyenv manager

### LINKED ISSUE #9757: Add pyenv (.python-version) support
**What would you like Renovate to be able to do?**

Renovate should bump Python versions in `.python-version` files just like it does for Node versions in `.node-version` files.

**Did you already have any implementation ideas?**

I'll start a `pyenv` manager implementation based on the current `nodenv` manager, however other contributors are welcome to help as well.

### PR DESCRIPTION
**Pull request by caiofbpa**
## Changes:

This PR adds support for updating `.python-version` files which are used by `pyenv` to determine current Python version.

## Context:

Renovate updates python versions on `Dockerfile` but not on `.python-version`, causing possible mismatch between development and production versions.

Closes #9757 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by caiofbpa** [comment]
Resolve #9757

File: lib/manager/pyenv/extract.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as datasourceDocker from '../../datasource/docker';
+import type { PackageDependency, PackageFile } from '../types';
+
+export function extractPackageFile(content: string): PackageFile {
+  const dep: PackageDependency = {
+    depName: 'python',
+    currentValue: content.trim(),
+    datasource: datasourceDocker.id,
+    lookupName: 'python',
```
Comment (viceice) [inline lib/manager/pyenv/extract.ts:9]:
```suggestion
```
Not required

File: lib/manager/pyenv/extract.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as datasourceDocker from '../../datasource/docker';
+import type { PackageDependency, PackageFile } from '../types';
+
+export function extractPackageFile(content: string): PackageFile {
+  const dep: PackageDependency = {
+    depName: 'python',
+    currentValue: content.trim(),
+    datasource: datasourceDocker.id,
```
Comment (viceice) [inline lib/manager/pyenv/extract.ts:8]: I think docker is a bad datasource, where did pyenv gets the python version from?

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by viceice** [comment]
@rarkins maybe we should use our python prebuild from containerbase as datasource? It's a little slower than normal python releases.

File: lib/manager/pyenv/extract.ts (hunk 2 above)
Comment (caiofbpa) [reply lib/manager/pyenv/extract.ts:8]: https://github.com/renovatebot/renovate/issues/9757#issuecomment-827915385

**Comment by rarkins** [comment]
> @rarkins maybe we should use our python prebuild from containerbase as datasource? It's a little slower than normal python releases.

Can we work out if the Docker images are delayed anyway?

**Comment by caiofbpa** [comment]
Anyone can help me fix that `lib/manager/index.spec.ts#L39` failing test?

**Comment by viceice** [comment]
> Anyone can help me fix that `lib/manager/index.spec.ts#L39` failing test?

You need to add the new manager to the api.ts

**Comment by caiofbpa** [comment]
@viceice thanks for the guidance, now the tests should pass. Any direction regarding changing the Docker datasource to something else?

**Comment by caiofbpa** [comment]
Just a gentle ping on @viceice and @rarkins to check wether we can merge this as is, or what better datasource can be used for Python here as I have no idea

**Review by rarkins — APPROVED** [review]

**Pull request merged**